web/html/src/manager/virtualization/guests/list/guests-list.js

```diff
@@ -0,0 +1,402 @@
+// @flow
+const React = require("react");
+const PropTypes = React.PropTypes;
+const MessageContainer = require("components/messages").Messages;
+const MessagesUtils = require("components/messages").Utils;
+const {Table, Column, SearchField, Highlight} = require("components/table");
+const Functions = require("utils/functions");
+const Utils = Functions.Utils;
+const {LinkButton, AsyncButton} = require("components/buttons");
+const {DeleteDialog} = require("components/dialog/DeleteDialog");
+const {DangerDialog} = require("components/dialog/DangerDialog");
+const {ModalButton} = require("components/dialog/ModalButton");
+const Systems = require("components/systems");
+import VirtualizationGuestActionApi from '../virtualization-guest-action-api';
+import VirtualizationGuestsListRefreshApi from '../virtualization-guests-list-refresh-api';
+
+class GuestsList extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      actionsResults: {},
+      errors: [],
+      selectedItems: []
+    }
+
+    this.onBeforeUnload = (event) => {
+        this.setState({
+            pageUnloading: true
+        });
+    };
+
+    this.handleSelectItems = (items) => {
+      this.setState({
+        selectedItems: items
+      });
+    };
+
+    this.selectGuest = (row) => {
+      this.setState({
+        selected: row
+      });
+    };
+
+    this.actionCallback = (results) => {
+      var newActions = this.state.actionsResults;
+      Object.entries(results).forEach(([uuid, actionid]) => {
+        newActions[uuid] = {'id': actionid, 'result': 'Queued'};
+      });
+      this.setState({actionsResults: Object.assign({}, this.state.actionsResults, newActions)});
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.salt_entitled) {
+      const port = window.location.port;
+      const url = "wss://" +
+        window.location.hostname +
+        (port ? ":" + port : "") +
+         "/rhn/websocket/minion/virt-notifications";
+      const ws = new WebSocket(url);
+
+      ws.onopen = () => {
+        // Tell the websocket that we want to hear from all action results on this virtual host.
+        ws.send(this.props.server_id);
+      }
+
+      ws.onclose = (e) => {
+        const errs = this.state.errors ? this.state.errors : [];
+        if (!this.state.pageUnloading && !this.state.websocketErr) {
+            errs.push(t("Websocket connection closed. Refresh the page to try again."));
+        }
+        this.setState({
+            errors: errs,
+        });
+      };
+
+      ws.onerror = (e) => {
+        console.error("Websocket error: " + e);
+        this.setState({
+           errors: [t("Error connecting to server. Refresh the page to try again.")],
+           websocketErr: true
+        });
+      };
+
+      ws.onmessage = (e) => {
+        const event = JSON.parse(e.data);
+        var newActions = this.state.actionsResults;
```

Comment:
this should be `var newActions = {}`

Reply:
Ok, I'll change them on Monday morning: no need to hurry now